sklearn/model_selection/_search.py

```diff
@@ -532,25 +532,51 @@ def inverse_transform(self, Xt):
         self._check_is_fitted('inverse_transform')
         return self.best_estimator_.transform(Xt)
 
-    def _fit(self, X, y, groups, parameter_iterable):
-        """Actual fitting,  performing the search over parameters."""
+    @property
+    def _param_iterable(self):
+        """To generate parameter iterables for multiple iterations"""
+        if hasattr(self, 'param_grid'):
+            return ParameterGrid(self.param_grid)
+        else:
+            return ParameterSampler(
+                self.param_distributions, self.n_iter,
+                random_state=self.random_state)
+
+    def fit(self, X, y=None, groups=None):
+        """Run fit with all sets of parameters.
+
+        Parameters
+        ----------
 
+        X : array-like, shape = [n_samples, n_features]
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples] or [n_samples, n_output], optional
+            Target relative to X for classification or regression;
+            None for unsupervised learning.
+
+        groups : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+        """
         estimator = self.estimator
         cv = check_cv(self.cv, y, classifier=is_classifier(estimator))
         self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
 
         X, y, groups = indexable(X, y, groups)
         n_splits = cv.get_n_splits(X, y, groups)
-        if self.verbose > 0 and isinstance(parameter_iterable, Sized):
-            n_candidates = len(parameter_iterable)
+        # Regenerate parameter iterable for each fit
+        candidate_params = list(self._param_iterable)
```

Comment:
This stores all the dicts as a list but this should not cause any (additional) memory (compared to 0.18) as it is anyway being stored a couple of lines later. Infact since we do not return the parameters and store them at `out`, it should essentially reduce the memory consumption..